src/main/java/bookstore/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,107 @@
+package bookstore.service;
+
+import bookstore.dto.order.OrderDto;
+import bookstore.dto.order.OrderRequestDto;
+import bookstore.dto.order.OrderStatusRequestDto;
+import bookstore.dto.orderitem.OrderItemDto;
+import bookstore.exception.EntityNotFoundException;
+import bookstore.mapper.OrderItemMapper;
+import bookstore.mapper.OrderMapper;
+import bookstore.model.Order;
+import bookstore.model.OrderItem;
+import bookstore.model.ShoppingCart;
+import bookstore.model.User;
+import bookstore.repository.OrderItemRepository;
+import bookstore.repository.OrderRepository;
+import bookstore.repository.ShoppingCartRepository;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    public List<OrderDto> findAll(Pageable pageable, Authentication authentication) {
+        Long userId = ((User) authentication.getPrincipal()).getId();
+        return orderMapper.toOrderDtoList(orderRepository.findAllByUserId(userId, pageable));
+    }
+
+    @Override
+    public OrderDto save(OrderRequestDto requestDto, Authentication authentication) {
+        User user = (User) authentication.getPrincipal();
+        Order order = new Order();
+        order.setUser(user);
+        order.setStatus(Order.Status.PENDING);
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(requestDto.getShippingAddress());
+
+        Set<OrderItem> orderItems = new HashSet<>();
+        BigDecimal total = BigDecimal.ZERO;
+
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId());
+        if (shoppingCart.getCartItems() == null) {
+            throw new NullPointerException("Shopping cart is empty");
```

Comment:
bad practice to throw NullPointerException
```suggestion
            throw new OrderProcessingException("Shopping cart is empty");
```

also you can provide userId in message like "Cart is empty for user: " + userId